packages/design-system/src/components/Status/Primitive/StatusPrimitive.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import classnames from 'classnames';
+import { IconName } from '@talend/icons';
+import Tooltip from '../../Tooltip';
+import { Icon } from '../../Icon/Icon';
+import Loading from '../../Loading';
+import { StackHorizontal } from '../../Stack';
+
+import styles from './Status.module.scss';
+
+export const variants = {
+	successful: 'successful',
+	failed: 'failed',
+	inProgress: 'inProgress',
+	warning: 'warning',
+	canceled: 'canceled',
+};
+
+export type StatusProps = {
+	icon?: IconName;
+	inProgress?: boolean;
+	hideText?: boolean;
+	children?: string;
+	variant: keyof typeof variants;
+};
+
+const Status = React.forwardRef(
+	(
+		{ children, icon, inProgress, hideText, variant, ...rest }: StatusProps,
+		ref: React.Ref<HTMLSpanElement>,
+	) => {
+		const text = <span className={styles.status__text}>{children}</span>;
+		const picto = (
+			<span className={styles.status__icon} aria-hidden>
+				{!inProgress && icon && <Icon name={icon} />}
+				{inProgress && <Loading />}
+			</span>
+		);
+
+		return (
+			<span {...rest} className={classnames(styles.status, styles[variant])} ref={ref}>
+				<StackHorizontal as="span" display="inline" gap="XXS" align="center" justify="start">
```

Comment:
Yes. I'm ok with this because IMO the two items represent two different things: 
The first span holds the responsibilities of the Stack's core styles. 
The second one (output of Stack) holds the layout. 

There is a way around that but it involves opening up `StackHorizontal` to classNames, which is something we're very cagey about.

Reply:
The two nested spans aren't that bad IMO.
The styles are different so there's a purpose.
I think the tradeoff of opening `StackHorizontal`  isn't worth it.